src/nvim/eval.c

```diff
@@ -22633,14 +22634,15 @@ static inline void term_delayed_free(void **argv)
   term_job_data_decref(j);
 }
 
+/// Called when the terminal buffer gets wiped while the job is still running
 static void term_close(void *d)
 {
   TerminalJobData *data = d;
   if (!data->exited) {
     data->exited = true;
```

Comment:
Looks like we are missing here a `term_job_data_decref(data);`.

If the terminal job exits first, two calls to `term_job_data_decref()` are made.

If the buffer is wiped first, only one call to `term_job_data_decref()` is made, because
the flag `data->exited` prevents the terminal job callback `eval_job_process_exit()` to call `terminal_exit() -> free_cb()`.